File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hash= new HashMap<>();
        int[] ans= new int[2];

        for(int i=0;i<nums.length;i++){
            hash.put(target-nums[i],i);
        }

        for(int i=0;i<nums.length;i++){
            if(hash.containsKey(nums[i]) && i!= hash.get(nums[i])){
                ans[0]=i;
                ans[1]= hash.get(nums[i]);
            }
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        n= len(nums)

        for i in range(n):
            c= target - nums[i]
            if c in hashMap:
                return [hashMap[c],i]
            hashMap[nums[i]]=i
        return []